Vizzuality/landgriffon
# Feature: LANDGRIF-959-Add dynamic indicator coefficient validation (part 1)

### General description

This PR adds the first piece of the task:
[Calculate Impact only for active indicators](https://vizzuality.atlassian.net/jira/software/c/projects/LANDGRIF/boards/14?selectedIssue=LANDGRIF-959)

Adding a dynamic validator at controller level to validate that received coefficients for indicators in the intervention creation flow matches the active indicators in the database

Adds a new test file (intervention controller validation) aiming to separate different tests so it can be more readable and also easier to mock when we can actually avoid building heavy preconditions

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

To manually test this feature you'll need to set to true USE_NEW_METHODOLOGY env var and randomly set some indicators to true and others to false.

the scenarioIntervention.newIndicatorCoefficient should be validated agains the nameCodes of the active indicators.

See added tests for more ingo

- Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
